java/vector/src/main/java/org/apache/arrow/vector/UInt2Vector.java

```diff
@@ -308,6 +309,12 @@ public TransferPair makeTransferPair(ValueVector to) {
     return new TransferImpl((UInt2Vector) to);
   }
 
+  @Override
+  public void setEncodedValue(int index, int value) {
+    Preconditions.checkArgument(value <= 0xFFFF, "value is overflow:" + value);
```

Comment:
```suggestion
    Preconditions.checkArgument(value <= Character.MAX_VALUE, "value is overflow:" + value);
    
```
I think this should work here.

Reply:
fixed.